porting/npl/riot/include/nimble/nimble_npl_os.h

```diff
@@ -215,7 +215,7 @@ ble_npl_callout_is_active(struct ble_npl_callout *c)
 static inline ble_npl_time_t
 ble_npl_callout_get_ticks(struct ble_npl_callout *co)
 {
-    return co->target_ticks;
+    return (ble_npl_time_t)co->target_ticks;
```

Comment:
this looks a bit odd,  shouldn't ble_npl_time_t in riot port be change to uint64_t ?

Reply:
I agree that it is odd and there is a bug here.

I don't think it is necessary to change `ble_npl_time_t` to `uint64_t`. The main reason to change from `uint32_t` to `uint64_t` is in the places where NimBLE provides a time in milliseconds and RIOT expects it in microseconds. The conversion to microseconds should be paired with a conversion to `uint64_t` to compensate for the additional range provided by value as millisecond compared to microseconds.

The bug is this line is that `target_ticks` is not in milliseconds but in microseconds. The RIOT port specifies that [1 tick == 1 millisecond](https://github.com/apache/mynewt-nimble/blob/master/porting/npl/riot/include/nimble/nimble_npl_os.h#L248), so this function should also properly return a value in milliseconds and not in microseconds.

I prefer to stick to `uint32_t` for the `ble_npl_time_t` and make sure that values converted from microseconds to milliseconds wrap around at `2**32` milliseconds and not at `2**32/1000` milliseconds

Reply:
fair enough,  I'll leave final call to Hauke though :)